### PR TITLE
Use event which property rather than code property when possible

### PR DESCRIPTION
**Pull request by AlexandreSi**
Fixes Ctrl + Z on AZERTY layout that closes the project rather than undoing something

### CODE REVIEW

File: newIDE/app/src/KeyboardShortcuts/index.js [hunk 1]
```diff
@@ -19,6 +19,69 @@ type KeyType =
   | 'numrow-arith' // Numrow-, Numrow=
   | 'other'; // Tab, Space
 
+const eventWhichToCode = {
+  '48': 'Digit0',
```
Comment (ClementPasteau) [inline newIDE/app/src/KeyboardShortcuts/index.js:23]: what's below 48?

Comment (AlexandreSi) [reply newIDE/app/src/KeyboardShortcuts/index.js:23]: Check here:

https://www.freecodecamp.org/news/javascript-keycode-list-keypress-event-key-codes/

File: newIDE/app/src/KeyboardShortcuts/index.js [hunk 2]
```diff
@@ -19,6 +19,69 @@ type KeyType =
...
+  '123': 'F12',
+  '188': 'Comma',
+};
+
+const getCodeFromEvent = (e: KeyboardEvent): string => {
+  if (
+    (e.which >= 65 && e.which <= 90) ||
+    (e.which >= 48 && e.which <= 57) ||
+    (e.which >= 112 && e.which <= 123) ||
+    e.which === 188
```
Comment (ClementPasteau) [inline newIDE/app/src/KeyboardShortcuts/index.js:79]: could this be simplified by `if (e.which in eventWhichToCode)` ?
From my tests, this seems to work

**Review by ClementPasteau — APPROVED** [review]
lgtm!

**Review by 4ian — COMMENTED** [review]
I'm a bit late but is https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key not better than `which`?

**Comment by AlexandreSi** [comment]
> I'm a bit late but is https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key not better than `which`?

I started implementing with key but I feel like it brings complexity. For instance:
Ctrl + A gives `event.key = a`
Shift + A gives `event.key = A`
Alt + A gives `event.key = æ`

`which` is the property that is layout-dependent (`code` is not, see the tests with `M`, `,` and `;`) and simple enough at the moment for the shortcuts we need.

**Pull request merged**